data/pigui/modules/info-view/06-flightlog.lua

```diff
@@ -109,119 +136,92 @@ local function renderCustomLog()
-local function renderStationLog()
-	local counter = 0
-	local was_clicked = false
-	for systemp, deptime, money, entry in FlightLog.GetStationPaths() do
-		counter = counter + 1
-
-		local station_type = "FLIGHTLOG_" .. systemp:GetSystemBody().type
-		headerText(l.DATE, formatDate(deptime))
-		headerText(l.STATION, string.interp(l[station_type],
-			{ primary_info = systemp:GetSystemBody().name, secondary_info = systemp:GetSystemBody().parent.name }))
-		-- headerText(l.LOCATION, systemp:GetSystemBody().parent.name)
-		headerText(l.IN_SYSTEM, ui.Format.SystemPath(systemp))
-		headerText(l.ALLEGIANCE, systemp:GetStarSystem().faction.name)
-		headerText(l.CASH, Format.Money(money))
-
-		::input::
-		entering_text_station = inputText(entry, counter,
-			entering_text_station, FlightLog.UpdateStationEntry, "station", was_clicked)
-		ui.nextColumn()
+local Windows = {
+	exportButtonWindow = layout.NewWindow(l.LOG_EXPORT, nil, ui.WindowFlags {"AlwaysAutoResize", "NoFocusOnAppearing", "NoBringToFrontOnFocus", "NoSavedSettings"}),
+}
 
-		was_clicked = false
-		if ui.iconButton(icons.pencil, buttonSpaceSize, l.EDIT .. "##station"..counter) then
-			was_clicked = true
-			goto input
-		end
+local flightlogWindowsLayout = layout.New(Windows)
+flightlogWindowsLayout.mainFont = pionillium.medium
 
-		ui.nextColumn()
-		ui.separator()
-	end
-end
+Windows.exportButtonWindow.anchors = { ui.anchor.right,  ui.anchor.bottom } 
 
--- See comments on previous function
-local function renderSystemLog()
-	local counter = 0
-	local was_clicked = false
-	for systemp, arrtime, deptime, entry in FlightLog.GetSystemPaths() do
-		counter = counter + 1
-
-		headerText(l.ARRIVAL_DATE, formatDate(arrtime))
-		headerText(l.DEPARTURE_DATE, formatDate(deptime))
-		headerText(l.IN_SYSTEM, ui.Format.SystemPath(systemp))
-		headerText(l.ALLEGIANCE, systemp:GetStarSystem().faction.name)
-
-		::input::
-		entering_text_system = inputText(entry, counter,
-			entering_text_system, FlightLog.UpdateSystemEntry, "sys", was_clicked)
-		ui.nextColumn()
+--- start with this window collapsed
+Windows.exportButtonWindow:Collapse()
 
-		was_clicked = false
-		if ui.iconButton(icons.pencil, buttonSpaceSize, l.EDIT .. "##system"..counter) then
-			was_clicked = true
-			goto input
-		end
+function Windows.exportButtonWindow.ShouldShow()
+	return true
+end
 
-		ui.nextColumn()
-		ui.separator()
+function Windows.exportButtonWindow.Show()
+	local c
+	c,export_html = ui.checkbox(l.LOG_HTML, export_html)
+	ui.sameLine()
+	if ui.button(l.SAVE) then
+		FlightLogExporter.Export( getIncludedSet(), earliest_first, true, export_html)
 	end
 end
 
-local function displayLog(logFn)
+
+local function displayFilterOptions()
+	ui.spacing()
+	local c
+	local flight_log = true;
+
+	c,include_custom_log = ui.checkbox(l.LOG_CUSTOM, include_custom_log)
+	c,include_station_log = ui.checkbox(l.LOG_STATION, include_station_log)
+	c,include_system_log = ui.checkbox(l.LOG_SYSTEM, include_system_log)
```

Comment:
Note: this could be simplified to:
```suggestion
	c,includedSet.Custom = ui.checkbox(l.LOG_CUSTOM, includedSet.Custom)
	c,includedSet.Station = ui.checkbox(l.LOG_STATION, includedSet.Station)
	c,includedSet.System = ui.checkbox(l.LOG_SYSTEM, includedSet.System)
```

This would remove the need for the `getIncludedSet` function - simply define an `includedSet = { System = true, ... }` table instead of the `include_*_log` variables currently defined.